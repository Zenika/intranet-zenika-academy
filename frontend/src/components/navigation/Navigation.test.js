import React from 'react';
import TestUtils from 'react-dom/test-utils';
import { shallow, mount } from 'enzyme';
import NavigationBar from './Navigation';

describe('Navbar renders correctly', () => {
  const navbar = mount(<NavigationBar />);

  beforeEach(() => {
    navbar.instance();
  });

  test('it renders correctly', () => {
    expect(navbar).toMatchSnapshot();
  });
  test('Displays the connexion Modal when sign in button is clicked', () => {
    const siButton = navbar.find('.signInDesktop');
    siButton.simulate('click');

    expect(navbar).toMatchSnapshot();
    expect(navbar.find('#modalLogin').children()).toBeDefined();
  });
});

describe('Testing Class Methods components', () => {
  const navbar = shallow(<NavigationBar ev />);
  const navbarInstance = navbar.instance();
  test('Corresponding Class method should be called when sign in button is clicked', () => {
    const ev = true;
    jest.spyOn(navbarInstance, 'toggleModal');

    const siButton = navbar.find('.signInDesktop');
    siButton.simulate('click');

    expect(navbarInstance.toggleModal).toHaveBeenCalledWith(ev);
  });
});

describe("Navigation bar when the user isn't connected", () => {
  const navbar = shallow(<NavigationBar />);
  const navbarInstance = navbar.instance();
  const nav = <NavigationBar />;

  test('Navigation bar displays a sign in button', () => {
    expect(nav).toBeDefined();
    expect(TestUtils.isElement(nav)).toBe(true);
    expect(TestUtils.isElementOfType(nav, NavigationBar)).toBe(true);

    const rendered = TestUtils.renderIntoDocument(<NavigationBar />);
    const siButton = TestUtils.findRenderedDOMComponentWithClass(
      rendered,
      'signInDesktop',
    );

    expect(siButton.textContent).toEqual('Se connecter');
  });

  beforeEach(() => {
    navbarInstance.componentDidMount();
  });

  test('Shallow rendering', () => {
    expect(navbar.state('modalState')).toBe('');
    expect(navbar.state('loggedIn')).toBe(false);
  });

  test('State mutates after clicking on sign in button', () => {
    const siButton = navbar.find('.signInDesktop');
    siButton.simulate('click');

    expect(navbar.state('modalState')).toBe(true);
    expect(navbar.state('loggedIn')).toBe(false);
  });
});
