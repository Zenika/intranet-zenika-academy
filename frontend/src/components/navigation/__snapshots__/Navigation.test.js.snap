// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly Displays the connexion Modal when sign in button is clicked 1`] = `
<NavigationBar>
  <nav
    aria-label="main navigation"
    className="navbar is-primary"
    role="navigation"
  >
    <div
      className="navbar-item display-mobile"
    >
      <button
        className="button"
        onClick={[Function]}
        type="submit"
      >
        Se connecter
      </button>
    </div>
    <div
      className="navbar-menu"
    >
      <div
        className="navBarLoggedOut"
      >
        <div
          className="navbar-start"
        >
          <p
            lang="en"
          >
            There is no fate but what we make
          </p>
        </div>
      </div>
      <div
        className="navbar-end"
      >
        <div
          className="navbar-item"
        >
          <button
            className="button signInDesktop"
            onClick={[Function]}
            type="submit"
          >
            Se connecter
          </button>
        </div>
      </div>
    </div>
    <SignInModal
      connect={[Function]}
      toggleModal={[Function]}
    >
      <section
        className="modal is-active"
        id="modalLogin"
      >
        <section
          className="modal-background"
        />
        <section
          className="modal-card"
        >
          <header
            className="modal-card-head"
          >
            <p
              className="modal-card-title"
            >
              Connexion
            </p>
          </header>
          <section
            className="modal-card-body"
          >
            <form>
              <section
                className="field"
              >
                <p
                  className="control has-icons-left"
                >
                  <input
                    className="input"
                    id="emailInput"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    required={true}
                    type="email"
                    value=""
                  />
                  <span
                    className="icon is-small is-left"
                  />
                </p>
              </section>
              <section
                className="field"
              >
                <p
                  className="control has-icons-left"
                >
                  <input
                    className="input"
                    id="passwordInput"
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    required={true}
                    type="password"
                    value=""
                  />
                  <span
                    className="icon is-small is-left"
                  >
                    <i
                      className="fas fa-lock"
                    />
                  </span>
                </p>
              </section>
              <footer
                className=" modalFooter"
              >
                <button
                  className="button is-success"
                  id="submitButton"
                  onClick={[Function]}
                  type="submit"
                >
                  Se connecter
                </button>
                <button
                  className="button"
                  id="cancelButton"
                  onClick={[Function]}
                  type="button"
                >
                  Annuler
                </button>
              </footer>
            </form>
          </section>
        </section>
      </section>
    </SignInModal>
  </nav>
</NavigationBar>
`;

exports[`Navbar renders correctly it renders correctly 1`] = `
<NavigationBar>
  <nav
    aria-label="main navigation"
    className="navbar is-primary"
    role="navigation"
  >
    <div
      className="navbar-item display-mobile"
    >
      <button
        className="button"
        onClick={[Function]}
        type="submit"
      >
        Se connecter
      </button>
    </div>
    <div
      className="navbar-menu"
    >
      <div
        className="navBarLoggedOut"
      >
        <div
          className="navbar-start"
        >
          <p
            lang="en"
          >
            There is no fate but what we make
          </p>
        </div>
      </div>
      <div
        className="navbar-end"
      >
        <div
          className="navbar-item"
        >
          <button
            className="button signInDesktop"
            onClick={[Function]}
            type="submit"
          >
            Se connecter
          </button>
        </div>
      </div>
    </div>
  </nav>
</NavigationBar>
`;
